framework/src/animation.cpp

```diff
@@ -30,7 +31,24 @@ namespace gvk
 					std::tie(rpos1, rpos2) = find_positions_in_keys(anode.mRotationKeys, timeInTicks);
 				}
 				auto rf = get_interpolation_factor(anode.mRotationKeys[rpos1], anode.mRotationKeys[rpos2], timeInTicks);
+
+#if 0
+				// original implementation (glm linear interpolation) - jerky
 				auto rotation = glm::lerp(anode.mRotationKeys[rpos1].mValue, anode.mRotationKeys[rpos2].mValue, rf);
+#elif 0
+				// glm spherical interpolation via mix - still jerky
+				auto rotation = glm::mix(anode.mRotationKeys[rpos1].mValue, anode.mRotationKeys[rpos2].mValue, rf);
+#elif 1
+				// using Assimp's interpolation - this fixes the jerks completely!
+				aiQuaternion q;
+				aiQuaternion::Interpolate(q, to_aiQuaternion(anode.mRotationKeys[rpos1].mValue), to_aiQuaternion(anode.mRotationKeys[rpos2].mValue), rf);
+				glm::quat rotation = to_quat(q);
+				// Note: not sure if normalizing the quaternion afterwards is actually necessary / improves anything. Looks good even without normalizing.
+#elif 0
+				// glm spherical interpolation via slerp - this looks good too, even without post-normalization
+				auto rotation = glm::slerp(anode.mRotationKeys[rpos1].mValue, anode.mRotationKeys[rpos2].mValue, rf);
+#endif
+				rotation=glm::normalize(rotation); // by itself, this fixes some of the jerks with glm lerp/mix interpolations, but not all of them
```

Comment:
I would prefer *not to* have debug-code in the `master` branch. Would you mind implementing only one solution here and keep the other options on a separate implementation-branch (which you could rebase on `master` from time to time). 

I'd prefer the `glm::slerp`-way over the `aiQuaternion`-way because then this code would be completely independent from ASSIMP which can enable usages that do not necessarily rely on ASSIMP in the future. Does `glm::slerp` get rid of all the artifacts in the same way as `aiQuaternion` does? The normalization-step is okay. Correctness is more important than correctness. 

One more thing: Do the to-be-slerped quaternions also have to be normalized? I.e. would it maybe make sense to apply a `glm::normalize`-step before storing the quaternions into `mRotationKeys`?! Probably yes?

Reply:
Tested a couple of models, found no glitches with `glm::slerp`. Will keep that variant.
Found no difference with/without normalizing the rotation keys in my models, but normalizing cannot hurt, just to be on the safe side.